packages/backend/src/open_payments/payment/incoming/service.ts

```diff
@@ -135,7 +141,35 @@ async function createIncomingPayment(
     data: incomingPayment.toData(0n)
   })
 
-  return await addReceivedAmount(deps, incomingPayment, BigInt(0))
+  incomingPayment = await addReceivedAmount(deps, incomingPayment, BigInt(0))
+  if (!deps.config.pollIncomingPaymentCreatedWebhook) {
+    return incomingPayment
+  }
+
+  try {
+    const response = await poll({
+      request: async () =>
+        getApprovedOrCanceledIncomingPayment(deps, { id: incomingPayment.id }),
+      pollingFrequencyMs: deps.config.incomingPaymentCreatedPollFrequency,
+      timeoutMs: deps.config.incomingPaymentCreatedPollTimeout
+    })
+
+    if (response) return response
```

Comment:
should we return the payment if it was cancelled?

Reply:
Always to return payment? How will end user now that the IP was cancelled? With other calls?

Reply:
I think for now lets only return the payment if it was approved, otherwise error. Since it might be confusing returning a payment that money can't be sent to at all.

Maybe in the future we can let the ASE specify what error is shown if a payment is not approved